### DISCUSSION TITLE
How to revalidate all form fields when blur event happens on any field?

### DISCUSSION THREAD

**Discussion by pavelspichonak** [react-hook-form/react-hook-form#8944] [Q&A] [answered]
I'm using **yup** validation and have **password** and **confirm password** fields [like in this example](https://stackblitz.com/edit/react-hook-form-confirm-password-match-validation?file=index.js)

I would like to revalidate **all** form fields when **blur** happens on **any** field. When **confirm password** field is invalid(has validation error) and we change **password** field I would like all form fields to be revalidated. But I do not know how to implement it. Looks like **mode** and **reValidateMode** properties allow to revalidate on **blur** event only **one** field which triggers **blur** event but does not allow to revalidate all form fields.

How can I revalidate **all** form fields on **blur** event when **any** field triggers **blur** event?

Thanks in advance!

**Comment by Moshyfawn** [comment, marked as answer]
For the password + password confirm example, there's a `deps` prop you want to provide to re-validate the dependant fields.
```
register("password", { deps: "confirmPassword" })
```

The `deps` API is a convenience wrapper around [the `trigger` method](https://react-hook-form.com/api/useform/trigger). So, in case of more complex form structure, you might want to opt-in to use [the `trigger` method](https://react-hook-form.com/api/useform/trigger) instead.

**Reply by pavelspichonak** [reply]
@Moshyfawn thanks! I'll read and try.

**Reply by Moshyfawn** [reply]
Sounds good!

**Reply by pavelspichonak** [reply]
I've read about this. Unfortunately I'm using [usecontroller](https://react-hook-form.com/api/usecontroller) instead of **register** method and looks like **usecontroller** has no option **deps**. Looks like the only solution is to pass **onBlur** handler to **Input** component and in this handler to call **trigger** function like this:
`const handleBlur = () => trigger('confirmPassword')`

So complicated...

**Reply by Moshyfawn** [reply]
```js
useController({
  rules: {
    deps: 'confirmPassword'
  }
})
```

**Reply by pavelspichonak** [reply]
thanks one more time

**Reply by Moshyfawn** [reply]
No problem!
